### PR TITLE
Supermatter events are now announced by "supermatter crystal" instead of "the supermatter crystal"

### LINKED ISSUE #27172: Supermatter Has Different Radio Chat Names For Anomaly Reporting and Integrity Reporting
### Exploit Reports

- [x] I confirm this issue is not an exploit. (Required)

### BYOND Version

514.1642

### Issue Description

In radio chat, the supermatter crystal is called "the supermatter crystal" when announcing anomalies, but is called "supermatter crystal" when reporting on an ongoing delamination.

### What did you expect to happen?

For the supermatter to have a single consistent name.

### What happened instead?

It has one name for anomaly reporting, and one for integrity reporting.

### Why is this bad/What are the consequences?

Inconsistency is bad.

Saboteurs need to remember which of the two names they want to mute on telecoms if they want to silence the crystal's cries for help or enter both.

### Steps to reproduce the issue.

Activate the supermatter, wait for anomaly to occur.
Delaminate it.

Notice the names it uses in radio chat are different.

### When did the problem start happening?

Probably since SM anomaly events got added.

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by Truls20**
## What Does This PR Do

Fixes #27172

## Why It's Good For The Game

Consistency is good, and you do not have to mute two different supermatter crystals to mute it in comms.

## Images of changes

image

## Testing
I caused a supermatter event, checked it was announced by "supermatter crystal". it was. 
I then caused a supermatter integrity alert to check if it was announced by "supermatter crystal", it was.
I then  caused a supermatter event in a supermatter shard to check if it was announced by "supermatter shard", it was. 
I then caused the supermatter shard to lose integrity to check if the integrity alert was announced by "supermatter shard", it was.
<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl:
fix: Fixed supermatter radio name inconsistency for both supermatter crystals and supermatter shards
/:cl:

### CODE REVIEW

**Comment by Truls20** [comment]
It seems the simple solution sadly leaves the supermatter **shards** inconsistent in radio. Not desirable, I shall find a better way to fix both

File: code/modules/power/engines/supermatter/supermatter_event.dm [hunk 1]
```diff
@@ -42,12 +42,12 @@
 /datum/supermatter_event/proc/sm_radio_say(text)
 	if(!text)
 		return
-	supermatter.radio.autosay(text, supermatter, "Engineering")
+	supermatter.radio.autosay(text, "supermatter crystal", "Engineering")
```
Comment (Contrabang) [inline code/modules/power/engines/supermatter/supermatter_event.dm:45]:
```suggestion
	supermatter.radio.autosay(text, supermatter.name, "Engineering")
```

File: code/modules/power/engines/supermatter/supermatter_event.dm [hunk 2]
```diff
@@ -42,12 +42,12 @@
...
 
 /datum/supermatter_event/proc/general_radio_say(text)
 	if(!text)
 		return
-	supermatter.radio.autosay(text, supermatter, null)
+	supermatter.radio.autosay(text, "supermatter crystal", null)
```
Comment (Contrabang) [inline code/modules/power/engines/supermatter/supermatter_event.dm:50]:
```suggestion
	supermatter.radio.autosay(text, supermatter.name, null)
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
This is how the normal supermatter autosay is done in supermatter.dm

**Review by Contrabang — APPROVED** [review]

**Review by Henri215 — APPROVED** [review]

**Review by DGamerL — APPROVED** [review]

**Pull request merged**